spec/features/update_existing_work_spec.rb

```diff
@@ -49,4 +49,32 @@
       expect(page).to have_content "title of deposit modified, abstract modified", count: 1
     end
   end
+
+  context "when existing work has many files" do
+    let(:work_version) { create(:work_version_with_work, :with_files, collection:, owner: user, title: original_title) }
+
+    before do
+      allow(Settings).to receive(:max_upload_files).and_return(1)
+    end
+
+    it "updates the work keeping attached files" do
+      visit work_path(work_version.work)
+
+      click_link "Edit #{original_title}"
+
+      expect(page).to have_content("You have more than 1 files in your deposit.")
+
+      # update the title and abstract
+      fill_in "Title of deposit", with: new_title
+      fill_in "Abstract", with: "I did really cool stuff"
+
+      click_button "Save as draft"
```

Comment:
since andrew's ticket mentioned that the draft saves, but without any files, i'm wondering if this could be changed to `click_button "Deposit"` and/or file count could be checked in the page expectations after form submission?  feels to me like one or both of those changes would more precisely check for the buggy behavior being fixed in this PR.

Reply:
In this test, the number of max upload files was set to 1 (instead of 250) and then a work version with 2 files was edited so that this test condition was triggered (editing a deposit with more than max files.)

In the expects below, the presence of both files is verified. I will add a comment to make that clearer.